File: Hearthstone-Script-Card-SDK/src/main/kotlin/club/xiaojiawei/bean/Card.java
package club.xiaojiawei.bean;

import club.xiaojiawei.CardAction;
import club.xiaojiawei.bean.area.Area;
import javafx.beans.property.ObjectProperty;
import javafx.beans.property.SimpleObjectProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.extern.slf4j.Slf4j;


/**
 * @author 肖嘉威
 * @date 2022/11/27 14:56
 */
@EqualsAndHashCode(callSuper = true)
@Data
@Slf4j
public class Card extends BaseCard{

    private CardAction action;

    private final ObjectProperty<Area> area = new SimpleObjectProperty<>();

    public Area getArea() {
        return area.get();
    }

    public ObjectProperty<Area> areaProperty() {
        return area;
    }

    public void setArea(Area area) {
        this.area.set(area);
    }

}

File: Hearthstone-Script-Card-SDK/src/main/kotlin/club/xiaojiawei/bean/Card.kt
package club.xiaojiawei.bean

import club.xiaojiawei.CardAction
import club.xiaojiawei.bean.area.Area
import club.xiaojiawei.mapper.BaseCardMapper
import javafx.beans.property.ObjectProperty
import javafx.beans.property.SimpleObjectProperty

/**
 * @author 肖嘉威
 * @date 2022/11/27 14:56
 */
class Card(var action: CardAction) : BaseCard(), Cloneable {

    val areaProperty: ObjectProperty<Area?> = SimpleObjectProperty()

    /**
     * 模拟用
     */
    var attackCount: Int = 0

    /**
     * 卡牌所在区域：手牌区、战场区等
     */
    var area: Area?
        get() = areaProperty.get()
        set(value) {
            areaProperty.set(value)
        }

    fun resetExhausted() {
        isExhausted = false
        attackCount = 0
    }

    @Override
    public override fun clone(): Card {
        try {
            val card = Card(this.action)
            BaseCardMapper.INSTANCE.update(this, card)
            return card
        } catch (e: CloneNotSupportedException) {
            throw RuntimeException(e)
        }
    }
}
